processor/transformer/transformer.go

```diff
@@ -350,31 +353,32 @@ func (trans *HandleT) request(ctx context.Context, url string, data []Transforme
 	}
 
 	var transformerResponses []TransformerResponseT
-	if statusCode == http.StatusOK {
-		integrations.CollectIntgTransformErrorStats(respData)
-
-		trace.Logf(ctx, "Unmarshal", "response raw size: %d", len(respData))
-		trace.WithRegion(ctx, "Unmarshal", func() {
-			err = jsonfast.Unmarshal(respData, &transformerResponses)
-		})
-		// This is returned by our JS engine so should  be parsable
-		// but still handling it
-		if err != nil {
-			trans.logger.Errorf("Data sent to transformer : %v", string(rawJSON))
-			trans.logger.Errorf("Transformer returned : %v", string(respData))
-			respData = []byte(fmt.Sprintf("Failed to unmarshal transformer response: %s", string(respData)))
-			transformerResponses = nil
-			statusCode = 400
-		}
-	}
 
 	if statusCode != http.StatusOK {
 		for i := range data {
 			transformEvent := &data[i]
 			resp := TransformerResponseT{StatusCode: statusCode, Error: string(respData), Metadata: transformEvent.Metadata}
 			transformerResponses = append(transformerResponses, resp)
 		}
+		return transformerResponses
 	}
+
+	integrations.CollectIntgTransformErrorStats(respData)
+
+	trace.Logf(ctx, "Unmarshal", "response raw size: %d", len(respData))
+	trace.WithRegion(ctx, "Unmarshal", func() {
+		err = jsonfast.Unmarshal(respData, &transformerResponses)
```

Comment:
Why are doing an unmarshal when the status code isn't 200. This would mean that the response wasn't sent by transformer and there is no need for it to be of `TransformerResponseT` type

Reply:
I think you missed the `}` in line 364. So, we are handling non-200 case before and this part of the code is for 200. Essentially this unmarshal is done for 200 cases.